### PR TITLE
feat: some detections for Azure.SignIn log type

### PR DESCRIPTION
**Pull request by edyesed**
### Background

This establishes a Pack for Azure.SignIn LogType detections and packages two detection for those logs.
1. A High Number of Failed Logins ( 10 ) in a short amount of time ( 10 minutes )
2. Users that authenticate with Legacy authentication protocols ( these users are high-risk as they aren't subject to complex authentication rules ). 

### Testing

```text
Azure.SignIn.ManyFailedSignIns
	[PASS] Failed Sign-In
		[PASS] [rule] true
		[PASS] [title] AzureSignIn: Multiple Failed LogIns for Principal [some-service-principal]
		[PASS] [dedup] some-service-principal
		[PASS] [alertContext] {"tenantId": "a2aa49aa-2c0c-49d2-af87-f402c421df0b", "source_ip": "12.12.12.12", "actor_user": "some-service-principal", "resourceDisplayName": "Azure Storage", "resourceId": "037694de-8c7d-498d-917d-edb650090fa5"}
	[PASS] Successful Sign-In
		[PASS] [rule] false

Azure.SignIn.LegacyAuth
	[PASS] Failed Sign-In
		[PASS] [rule] false
	[PASS] Sign-In with legacy auth
		[PASS] [rule] true
		[PASS] [title] AzureSignIn: Principal [homer.simpson@springfield.org] authenticated with a legacy auth protocol
		[PASS] [dedup] homer.simpson@springfield.org
		[PASS] [alertContext] {"tenantId": "c5fd010a-2972-4be2-9c67-986ceca2a042", "source_ip": "12.12.12.12", "actor_user": "homer.simpson@springfield.org", "resourceDisplayName": "Office 365 Exchange Online", "resourceId": "b54c474b-4d74-4920-b0c3-7a8d2f2c4e95", "userAgent": null}
	[PASS] Sign-In with legacy auth, KNOWN_EXCEPTION
		[PASS] [rule] false

--------------------------
Panther CLI Test Summary
	Path: .
	Passed: 2
	Failed: 0
	Invalid: 0

```

### CODE REVIEW

File: rules/azure_signin_rules/azure_failed_signins.yml [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+Filename: azure_failed_signins.py
+RuleID: "Azure.SignIn.ManyFailedSignIns"
+DisplayName: "Azure Many Failed SignIns"
+Enabled: true
+# Ten Failed Sign-Ins(Threshold) in Ten Minutes(DedupPeriodMinutes)
+Threshold: 10
+DedupPeriodMinutes: 10
+LogTypes:
+  - Azure.SignIn
+Severity: Low
```
Comment (andrea-youwakim) [inline rules/azure_signin_rules/azure_failed_signins.yml:11]: if we set the threshold to ten do we still want the sev to be low?

File: rules/azure_signin_rules/azure_legacyauth.yml [hunk 2]
```diff
@@ -0,0 +1,298 @@
+AnalysisType: rule
+Filename: azure_legacyauth.py
+RuleID: "Azure.SignIn.LegacyAuth"
+DisplayName: "Azure SignIn via Legacy Authentication Protocol"
+Enabled: true
+Threshold: 1
+DedupPeriodMinutes: 10
+LogTypes:
+  - Azure.SignIn
+Severity: Medium
```
Comment (andrea-youwakim) [inline rules/azure_signin_rules/azure_legacyauth.yml:10]: this one can totes stay medium but i will suggest low in the event signing in with the legacy auth protocol is not an inherent call to action

**Review by andrea-youwakim — COMMENTED** [review]
minor comments left, adds new pack and detections for azure sign in events

File: rules/azure_signin_rules/azure_failed_signins.yml (hunk 1 above)
Comment (edyesed) [reply rules/azure_signin_rules/azure_failed_signins.yml]: yeah, that makes sense to me. Severity updated in latest commit

File: rules/azure_signin_rules/azure_legacyauth.yml (hunk 2 above)
Comment (edyesed) [reply rules/azure_signin_rules/azure_legacyauth.yml:10]: I expanded the runbook a bit to include some terminology from https://learn.microsoft.com/en-us/azure/active-directory/conditional-access/block-legacy-authentication that explains a bit about why legacy authentication is a security liability. 

I agree that the call to action could be more prescriptive. From the link above ( now also in the runbook ), MSFT makes a strong case for why legacy auth should be totally disabled. I can update the runbook to mention such if you think that makes sense. 

In the current commit, I've included a text blurb from MSFT about the liability of legacy authentication and a link to the msft.com material linked above.

**Review by andrea-youwakim — APPROVED** [review]
new azure sign in pack and helpful detections

**Pull request merged**